libgeos/libgeos.go

```diff
@@ -0,0 +1,145 @@
+package libgeos
+
+// Package libgeos provides a cgo wrapper around the GEOS (Geometry Engine,
+// Open Source) library. To use this package, you will need to install the
+// GEOS library.
+//
+// Its purpose is to provide functionality that has been implemented in GEOS,
+// but is not yet available in the simplefeatures library.
+//
+// This package can be used in two ways:
+//
+// 1. Many GEOS non-threadsafe handles can be created, and functionality used
+// via those handles. This is useful if many threads need to perform geometry
+// operations concurrently (each thread should use its own handle). This method
+// of accessing GEOS functionality allows parallelism, but is more difficult to
+// use.
+//
+// 2. The Global functions can be used, which share an unexported global
+// handle. Usage is serialised using a mutex, so these functions are safe to
+// use concurrently. This method of accessing GEOS functionaliy is easier,
+// although doesn't allow parallelism.
+//
+// The operations in this package ignore Z and M values if they are present.
+
+import (
+	"sync"
+
+	"github.com/peterstace/simplefeatures/geom"
+)
+
+var (
+	globalMutex  sync.Mutex
+	globalHandle *Handle
+)
+
+func getGlobalHandle() (*Handle, error) {
```

Comment:
I think this function guarantees if there is no error that the handle will be non-nil?

Reply:
Yes, that's correct.